docs: jira cloud documentation (#3494)
Unleash/unleash: website/docs/reference/integrations/jira-cloud-plugin-usage.md

---
title: Jira Cloud Integration - Usage
---

With the Unleash Jira Cloud Plugin you can create, view and manage Unleash feature toggles directly from a Jira issue.

The plugin also shows you current status of connected toggles.

## Using the plugin

To use the plugin, you'll need to create an issue or use an existing one. Once the issue is saved and you open up the issue panel, you'll be greeted with a button to activate the Unleash plugin for that issue.

> ![Jira Cloud - Button to activate Unleash plugin in Jira Issue](/img/jira-cloud-activate-unleash-panel-button.png)

Reviewer: What do you call that row of buttons in Jira? I suggest something like this, but I don't know what to call that row.

Anyway, maybe something like this?

```suggestion
<Figure caption="The Unleash Feature Flags button gets added to the top-level actions of new issues. Use that button to activate the Unleash plugin for that issue." img="/img/jira-cloud-activate-unleash-panel-button.png" />
```

Author: The section the plugin content ends up in is called a Jira Issue Panel. I can't find a name mentioned anywhere for that row however. Maybe just go with your suggestion here

### Connecting a toggle to an issue

> ![Jira Cloud - Button that opens modal to connect toggles to Jira Issue](/img/jira-cloud-issue-button.png)

Reviewer: What is this an image of? Is this what happens after you "activate the Unleash plugin" for this issue? or is this something else?

Is this correct?

```suggestion
<Figure caption="Once active, the Unleash plugin adds a button labeled 'connect toggle' to the issue." img="/img/jira-cloud-issue-button.png" />

```

Author: Yeah that seems to be a correct understanding. Activating unleash for the issue adds an issue panel section that at first will only contain that button

> Clicking this button opens a dialog where you first select an Unleash Project to connect toggle from,

Reviewer: This is a little unclear. what is "this button"?

Actually, I think the first of these suggested paragraphs could go before the first image. What do you think? 🤔 

Second, I think these steps could stand to be outlined as an ordered list. Thoughts?
```suggestion
Use the issue's "connect toggle" button to open a dialog. 

There are a few steps to connect a toggle:

1. Select the Unleash project that contains the toggle:

```

> ![Jira Cloud - Dropdown to select Unleash project](/img/jira-cloud-select-project-expanded.png)

Reviewer: Does it contain all projects? or only some?

```suggestion
![Jira Cloud: connect feature toggle form. The project selection dropdown contains all Unleash projects](/img/jira-cloud-select-project-expanded.png)
```

Reviewer 2: It's all projects, nice suggestion

> allowing you then to either connect an existing toggle
> 
> ![Jira Cloud - Dropdown to select existing toggle to connect](/img/jira-cloud-add-existing-toggle.png)
> 
> or if you flip the switch that says "Create new toggle", allows you to create a new toggle
> 
> ![Jira Cloud - Switch that opens the form to create and connect a new toggle](/img/jira-cloud-add-new-toggle.png)

Reviewer: You might need to jimmy the indentation around here to make the list render properly.

```suggestion
2. Use the "create new toggle" option to choose whether you want to add an existing toggle or create a new one.  

<Figure caption="When you add an existing toggle, use the toggle name select list to choose from existing toggles in the selected Unleash project." img="/img/jira-cloud-add-existing-toggle.png" />

<Figure caption="When you add a new toggle, you must give it a name and can choose to give it a description." img="/img/jira-cloud-add-new-toggle.png" />c
```

Author: Yup, I'll accept the changes and inspect the rendering after

### Toggle status as part of Issue view

> Once you've connected at least one toggle to an issue, our plugin will list the current status of each environment for

Reviewer: ```suggestion
Once you've connected at least one toggle to an issue, the Unleash plugin will list the current status of each environment for
```

that toggle. If the current user is allowed to edit the issue, they can also enable or disable environments directly
from Jira.

> ![Jira Cloud - Overview of toggle status in Jira Issue](/img/jira-cloud-toggle-status.png)

Reviewer: ```suggestion
<Figure caption="A Jira Cloud issue with a connected toggle. The Unleash feature flags section now shows the connected toggles along with the toggle's environments. There are controls to enable or disable the toggle in the development and production environments" img="/img/jira-cloud-toggle-status.png" />
```

### Change Requests

> When enabling or disabling a toggle, if Change Requests have been activated for the Unleash project and environment the toggle belongs to
> 
> ![Jira Cloud - A toggle for an environment has been clicked](/img/jira-cloud-change-request-clicked-toggle.png)
> 
> a dialog will open to alert the user that a change request will have to be created.

Reviewer: What happens if you already have an active change request for that project and env?

```suggestion
The plugin respects Unleash's [change requests](../change-requests.md). If change requests are turned on in the connected project and the selected environment, the plugin will ask whether you want to create a change request or not. 

If you already have an active change request for that project and that environment, the changes will be added to your existing change request.
```

Reviewer 2: Yep, this suggestion is spot on @thomasheartman, that's exactly what happens, thank you!

@daveleek It's your PR so I don't want to fiddle but I think we should take this suggestion

Author: lets take it!

> ![Jira Cloud - Modal opens for creating a new change request](/img/jira-cloud-change-request-dialog.png)

Reviewer: I'm thinking this image could probably be removed? what do you think?

```suggestion
```

> Clicking the "Yes" button creates a 
> Change Request in Unleash, and a confirmation will be displayed in the dialog along with a link to the new Change Request.
> 
> ![Jira Cloud - Confirmation message for the change request just created](/img/jira-cloud-change-request-confirmation.png)

Reviewer: ```suggestion
If you confirm that you would like to open a change request, then the plugin will create one for you and present a confirmation dialog.

<Figure caption="A dialog appears when the plugin creates a change request for you. The dialog contains a link directly to the newly created change request." img="/img/jira-cloud-change-request-confirmation.png" />
```

Author: Yeah good stuff!

When the Change Request has been reviewed and applied in Unleash, the toggle will show the requested state after the next refresh 
of the issue and toggle status page.

### Disconnecting toggle from Issue

> If a toggle is no longer relevant for your Jira Issue, you can disconnect it using the Disconnect toggle button (
> provided your user has edit rights on the issue)

Reviewer: ```suggestion
If a toggle is no longer relevant for your Jira Issue, you can disconnect it using the "disconnect toggle" button. This button is only available if your user has edit permissions for the Jira issue.
```

Reviewer 2: I'm actually not sure about this one, I agree the change makes a lot of sense but I don't know that's what happens. It's really dumb but to test this I need to get it into a prod env in Jira and to do that I need the Atlassian review to complete

Author: Maybe keep this one in until we know for sure

> ![Jira Cloud - Highlighting the button that disconnects a toggle](/img/jira-cloud-disconnect-toggle.png)

Reviewer: ```suggestion
![Jira Cloud: issue with a connected toggle. The 'disconnect toggle' button (highlighted) is displayed next to the toggle's name.](/img/jira-cloud-disconnect-toggle.png)
```

---
title: Jira Cloud Integration - Usage
---

> The Jira Cloud Plugin is part of Unleash Enterprise.

With the Unleash Jira Cloud Plugin you can create new feature toggles in Unleash from a Jira issue as well as link
existing toggles to Jira issues.

The plugin also shows you current status of connected toggles.

## Using the plugin

> To use the plugin, you'll need to create an issue or use an existing one. Once the issue is saved and you open up the Issue panel, you'll be greeted with a button to activate the Unleash plugin for that issue.

Reviewer 2: ```suggestion
To use the plugin, you'll need to create an issue or use an existing one. Once the issue is saved and you open up the issue panel, you'll be greeted with a button to activate the Unleash plugin for that issue.
```
